target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
I definitively have to disable the flash endianness workaround, as I want to keep the original OS layout. This means I don't have to enable the EBU bus endian swap.

I confirm that the PCIe bus works well, as I can configure wifi access point and connect to it (after installing wpad and hostapd).

What do you think about adding an option in the dts files which enables the possibility to disable the endianness workaround ? At the beginning of lantiq-flash driver, you can see this:

```
/*
 * The NOR flash is connected to the same external bus unit (EBU) as PCI.
 * To make PCI work we need to enable the endianness swapping for the address
 * written to the EBU. This endianness swapping works for PCI correctly but
 * fails for attached NOR devices. To workaround this we need to use a complex
 * map. The workaround involves swapping all addresses whilst probing the chip.
 * Once probing is complete we stop swapping the addresses but swizzle the
 * unlock addresses to ensure that access to the NOR device works correctly.
 */
```
Adding a possibility to disable addresses swap by adding an option in the device dts file might be a good idea.

Regarding the flash layout, I'll try to keep the bootmonitor. I'll make a post-install (or first-boot) script that updates the mtd with new size and crc.

For the sysupgrade image generation, I have the following config:

```
define Device/RS353
  DEVICE_PROFILE := RS353
  DEVICE_TITLE := Bintec RS353
  KERNEL := kernel-bin | append-dtb | lzma | lzma-loader | uImage lzma
  IMAGE_SIZE := 31232k
  DEVICE_PACKAGES := kmod-usb-dwc2 kmod-ath9k wpad-mini kmod-i2c-gpio kmod-swconfig wpad-mini kmod-ledtrig-usbdev
endef
TARGET_DEVICES += RS353
```
I needed to add the uImage header to properly detect the rootfs partition in the "firmware" mtd device. This doesn't affects the lzma-loader as I jump after the uImage header (which has always the same size).